lib/api/controllers/memoryStorageController.js

```diff
@@ -666,40 +836,71 @@ function extractArgumentsFromRequestForZAdd (request) {
  * @returns {*[]}
  */
 function extractArgumentsFromRequestForZInterstore (request) {
-  var args = [
-      request.input.body.destination
-    ],
-    keys = [];
+  let args = [request.input.resource._id];
 
-  if (request.input.resource._id) {
-    keys.push(request.input.resource._id);
-  }
-  if (request.input.body.keys !== undefined) {
-    assertBodyAttributeType(request, 'keys', 'array');
+  kassert.assertHasId(request);
+  kassert.assertHasBody(request);
+  kassert.assertBodyAttributeType(request, 'keys', 'array');
 
-    keys = keys.concat(request.input.body.keys);
+  if (request.input.body.keys.length === 0) {
+    throw new BadRequestError(`ms:${request.input.action} At least 1 source key must be provided`);
   }
 
-  args.push(keys.length);
-  args = args.concat(keys);
+  args.push(request.input.body.keys.length);
+  args.push(...request.input.body.keys);
```

Comment:
Should be prevented by request size limit. Even if it isn't, the exception will be catched and a proper error 500 will be returned to the client.

If we want to catch this, an arbitrary limit should be added. And not only to this API: every route taking an array should be checked. 

I don't feel like fixing this small issue right now :)